modules/platforms/dotnet/Apache.Ignite.Core/Impl/Plugin/PluginProcessor.cs

```diff
@@ -145,18 +149,40 @@ public ExceptionFactory GetExceptionMapping(string className)
 
             ExceptionFactory res;
 
-            return _exceptionMappings.TryGetValue(className, out res) ? res : null;
+            if (_exactExceptionMappings.TryGetValue(className, out res))
+            {
+                return res;
+            }
+
+            foreach (var mapping in _patternExceptionMappings)
+            {
+                if (className.StartsWith(mapping.Key))
+                {
+                    return mapping.Value;
+                }
+            }
+
+            return null;
         }
 
         /// <summary>
         /// Registers the exception mapping.
         /// </summary>
+        /// <param name="className">Full class name of java exception or class name pattern (if ends with '*').</param>
+        /// <param name="factory">Exception factory for matched java class name.</param>
         public void RegisterExceptionMapping(string className, ExceptionFactory factory)
         {
             Debug.Assert(className != null);
             Debug.Assert(factory != null);
 
-            _exceptionMappings.GetOrAdd(className, _ => factory);
+            if (className.EndsWith("*"))
+            {
+                _patternExceptionMappings.GetOrAdd(className.TrimEnd('*'), _ => factory);
```

Comment:
`className.Substring(0, className.Length - 1)` to avoid dealing with cultures.

Reply:
Thanks, fixed.